### PR TITLE
[RTC-231] move server notifications to protobuffs

### PR DESCRIPTION
**Pull request by Rados13**

### CODE REVIEW

**Review by sgfn — APPROVED** [review]
LGTM!

File: jellyfish/server_notifications.proto [hunk 1]
```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+package Jellyfish.Server;
+
```
[the remark this answers is not in the document]
Comment (Rados13) [reply jellyfish/server_notifications.proto:4]: If I understand accordingly to [Proto Best Practicies](https://protobuf.dev/programming-guides/dos-donts/):
```
Don’t Add a Required Field
Never add a required field, instead add // required to document the API contract. Required fields are considered harmful by so many they were removed from proto3 completely. Make all fields optional or repeated. You never know how long a message type is going to last and whether someone will be forced to fill in your required field with an empty string or zero in four years when it’s no longer logically required but the proto still says it is.
```
If we will want to extend `peerCrashed` with the reason we will have to add reason as an optional parameter, so we won't have to pass it to `peerConnected` and `peerDisconnected`, but maybe I am wrong.

[the remark this answers is not in the document]
Comment (mickel8) [reply jellyfish/server_notifications.proto:4]: Then why not to create one message and put there all possible fields?

[the remark this answers is not in the document]
Comment (LVala) [reply jellyfish/server_notifications.proto:4]: @Rados13 I don't see how what you wrote contradicts with @mickel8's comment -- when you want to add error message, you simply extend one of the messages, others are not affected. The documentation that you linked warns against making fields explicitly `required`, which you [cannot do using `proto3`](https://protobuf.dev/programming-guides/proto3/#specifying-rules).

[the remark this answers is not in the document]
Comment (LVala) [reply jellyfish/server_notifications.proto:4]: Btw, I agree with @mickel8 on that 🐟

File: jellyfish/server_notifications.proto [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    ServerNotification.PeerNotification peerCrashed = 4;
+    ServerNotification.ComponentNotification componentCrashed = 5;
+    ServerNotification.Authenticated authenticated = 6;
+
+    ClientMessage.TokenMessage authRequest=7;
+  }
+}
+
+message ClientMessage {
+  message TokenMessage{
```
[the remark this answers is not in the document]
Comment (LVala) [reply jellyfish/server_notifications.proto:19]: Also, protobufs use `snake_case` for field names, see [style guide](https://protobuf.dev/programming-guides/style/).

File: jellyfish/server_notifications.proto [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    message PeerNotification{
+        string roomId=1;
+        string peerId=2;
+    }
+
+    message ComponentNotification{
+        string roomId=1;
+        string componentId=2;
+    }
+}
```
Comment (LVala) [inline jellyfish/server_notifications.proto:40]: I don't know if using `ServerNotification` and `ClientMessage` only to group other `message` definitions is a good idea -- now in theory empty `ServerNotification` is a valid message. I get that you can validate if some message is actually `ServerNotification` or `ClientMessage` in Elixir code, but wouldn't it be more idiomatic to do
```protobuf
message ControlMessage {
  message ServerNotification {
    message PeerCrashed {
      string peer_id = 1;
      [...]
    }
    [...]
    oneof content {
      PeerCrashed peer_crashed = 1;
    }
  }

  message ClientMessage {
    [...]
  }
  oneof content {
    ServerNotification server_notification = 1;
    ClientMessage client_message = 2;
  }
}
```
Or something like that, doesn't have to be nested ofc. I'm not sure tho.

**Review by LVala — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -1,6 +1,12 @@
 # Protos
 
-This repository contains protocol buffers definitions for [jellyfish-dev organisation](https://github.com/jellyfish-dev) repositories.
+This repository contains protocol buffers definitions for [jellyfish-dev organization](https://github.com/jellyfish-dev) repositories.
+
+## How to add this repo as a git submodule in your own repository
+
+Use this command below:
```
Comment (mickel8) [inline README.md:7]:
```suggestion
Use the command below:
```

**Review by mickel8 — APPROVED** [review]

**Comment by mickel8** [comment]
Please remember to squash commits when merging

**Review by LVala — CHANGES_REQUESTED** [review]
Besides this style issue, it's 👍

**Review by LVala — APPROVED** [review]
👍

**Pull request merged**